docs/spec_draft.md

```diff
@@ -543,6 +544,124 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dot_general
+
+### Semantics
+
+Performs general dot products between vectors, vector/matrix and matrix/matrix
+multiplication over dimensions specified in `dot_dimension_numbers`.
+
+### Operands
+
+| Name                    | Type                                                                                                                                                                                                 |
+|-------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `lhs`                   | tensor of integer, floating-point or complex element types                                                                                                                                           |
+| `rhs`                   | tensor of integer, floating-point or complex element types                                                                                                                                           |
+| `dot_dimension_numbers` | attribute containing the following 1-dimensional array of `si64`:<br>* `lhs_batching_dimensions`<br>* `rhs_batching_dimensions`<br>*  `lhs_contracting_dimensions`<br>* `rhs_contracting_dimensions` |
+| `precision_config`      | attribute containing the following precision(s):<br>* `STABLEHLO_PRECISION_DEFAULT`<br>* `STABLEHLO_PRECISION_HIGH`<br>* `STABLEHLO_PRECISION_HIGHEST`                                               |
+
+### Results
+
+| Name     | Type                                                       |
+|----------|------------------------------------------------------------|
+| `result` | tensor of integer, floating-point or complex element types |
+
+### Constraints
+
+  * (C1) `lhs`, `rhs` and `result` have the same element type.
+  * (C2) `lhs` and `rhs` have the same number of `batching_dimensions`,
+  `contracting_dimensions`.
+  * (C3) `lhs_batching_dimensions` and `lhs_contracting_dimensions` values are
+  unique.
+  * (C4) `rhs_batching_dimensions` and `rhs_contracting_dimensions` values are
+  unique.
+  * (C5) `dim(lhs_batching_dimensions, i)`, `dim(rhs_batching_dimensions, i)`
```

Comment:
Can you elaborate what are we checking here?

`dim(lhs_batching_dimensions, i)` < `dim(lhs_contracting_dimensions, j)`

From the look it seems: The dimension size of lhs_batching_dimensions need to be less that the dimension sizes of contracting dimensions

Reply:
This constraint should be removed.